core/dbt/include/global_project/macros/etc/column_helpers.sql

```diff
@@ -0,0 +1,59 @@
+{% macro column_list(columns) %}
+  {%- for col in columns %}
+    {{ col.name }} {% if not loop.last %},{% endif %}
+  {% endfor -%}
+{% endmacro %}
+
+
+{% macro column_list_for_create_table(columns) %}
+  {%- for col in columns %}
+    {{ col.name }} {{ col.data_type }} {%- if not loop.last %},{% endif %}
+  {% endfor -%}
+{% endmacro %}
+
+{% macro get_quoted_csv(column_names) %}
+    
+    {% set quoted = [] %}
+    {% for col in column_names -%}
+        {%- do quoted.append(adapter.quote(col)) -%}
+    {%- endfor %}
+
+    {%- set dest_cols_csv = quoted | join(', ') -%}
+    {{ return(dest_cols_csv) }}
+
+{% endmacro %}
+
+
+{% macro diff_columns(source_columns, target_columns) %}
+
+  {% set result = [] %}
+  {% set source_names = source_columns | map(attribute = 'column') | list %}
+  {% set target_names = target_columns | map(attribute = 'column') | list %}
+   
+   {# --check whether the name attribute exists in the target - this does not perform a data type check #}
+   {% for sc in source_columns %}
+     {% if sc.name not in target_names %}
+        {{ result.append(sc) }}
+     {% endif %}
+   {% endfor %}
+  
+  {{ return(result) }}
+
+{% endmacro %}
```

Comment:
curious (but no strong opinion), on the home for helper macros that are only currently used for a specific materialization... I like the categorization but maybe unnecessary complexity to go to another file when no other macros are using it?

Reply:
Yeah, good thought. As above, the question for me is:
1. Should we consider these "private" methods that exist for internal (global materialization) use only, not to be documented, and used at users' own risk
2. Should dbt-core's global project become more of a global repository of handy, nice-to-have helper macros like this one
3. Somewhere in between: these are mostly internal methods, they're useful for adapter plugin maintainers when extending `dbt-core` to new databases, but they shouldn't be called / relied on any old project

When I put it like that, I actually find myself leaning toward 2 or 3, rather than 1. Which is to say, they should probably live in a file like `materializations/internal_helpers`